src/test/ui/suggestions/imm-ref-trait-object-literal-bound-regions.stderr

```diff
@@ -5,6 +5,7 @@ LL |     foo::<S>(s);
    |     ^^^^^^^^ the trait `for<'b> Trait` is not implemented for `&'b S`
    |
    = help: the trait `Trait` is implemented for `&'a mut S`
+   = note: `for<'b> Trait` is implemented for `&'b mut S`, but not for `&'b S`
```

Comment:
This is less than ideal, but I think it should be left for a separate PR.

Reply:
Yeah, but I was not really able to understand why line 7 is printing only not satisfied trait and a different name for the lifetime from the one use in the code (`'a` instead of `'b`) and note due to change is using binder and the same name for lifetime than the one in the code.